sql/moz-fx-data-shared-prod/adjust_derived/adjust_derived_v1/query.py

```diff
@@ -0,0 +1,305 @@
+            write_dict_to_csv(csv_data, f_csv.name)
+
+            client = bigquery.Client(project)
+
+            print("hey, I'm about to config the job\n")
+
+            job_config = bigquery.LoadJobConfig(
+                create_disposition="CREATE_IF_NEEDED",
+                write_disposition="WRITE_TRUNCATE",
+                time_partitioning=bigquery.TimePartitioning(
+                    type_=bigquery.TimePartitioningType.DAY,
+                    field="date",
+                ),
+                skip_leading_rows=1,
+                schema=[
+                    bigquery.SchemaField("date", "DATE"),
+                    bigquery.SchemaField("app", "STRING"),
+                    bigquery.SchemaField("app_token", "STRING"),
+                    bigquery.SchemaField("network", "STRING"),
+                    bigquery.SchemaField("network_token", "STRING"),
+                    bigquery.SchemaField("campaign", "STRING"),
+                    bigquery.SchemaField("campaign_token", "STRING"),
+                    bigquery.SchemaField("adgroup", "STRING"),
+                    bigquery.SchemaField("adgroup_token", "STRING"),
+                    bigquery.SchemaField("creative", "STRING"),
+                    bigquery.SchemaField("creative_token", "STRING"),
+                    bigquery.SchemaField("country", "STRING"),
+                    bigquery.SchemaField("os", "STRING"),
+                    bigquery.SchemaField("device", "STRING"),
+                    bigquery.SchemaField("clicks", "INTEGER"),
+                    bigquery.SchemaField("installs", "INTEGER"),
+                    bigquery.SchemaField("limit_ad_tracking_install_rate", "FLOAT"),
+                    bigquery.SchemaField("click_conversion_rate", "FLOAT"),
+                    bigquery.SchemaField("impression_conversion_rate", "FLOAT"),
+                    bigquery.SchemaField("sessions", "INTEGER"),
+                    bigquery.SchemaField("daus", "INTEGER"),
+                    bigquery.SchemaField("waus", "INTEGER"),
+                    bigquery.SchemaField("maus", "INTEGER"),
+                ],
+            )
+
+            # Table names are based on the app name seen in the Adjust dashboard"
+            destination = f"{project}.{dataset}.adjust_deliverables_v1_${partition}"
+            print(destination)
+
+            job = client.load_table_from_file(f_csv, destination, job_config=job_config)
+
+            print(f"Running job {job.job_id}")
+            job.result()
+
+
+def main():
+    """Input data, call functions, get stuff done."""
+    parser = ArgumentParser(description=__doc__)
+    parser.add_argument("--date", required=True)
+    parser.add_argument("--project", default="moz-fx-data-shared-prod")
+    parser.add_argument("--dataset", default="adjust_derived")
+
+    args = parser.parse_args()
+
+    app_list = json.loads(APP_TOKEN_LIST)
+
+    # Cycle through the apps to get the relevant kpi data
+    for num in range(len(app_list)):
```

Comment:
A more pythonic way of iterating through the app_list would be to do:
```
for app in app_list:
    [code to handle each app]
```
This will allow you to avoid having to access the app using the index from the whole object each time you need to access an app specific value:

```
[before]
print(
    f'This is data for {app_list[num]["app_name"]} - {app_list[num]["app_token"]}'"
)
[after]
print(
    f'This is data for {app_list["app_name"]} - {app_list["app_token"]}'"
)
```

Reply:
wait, wouldn't it be:

> print(
>     f'This is data for {app["app_name"]} - {app["app_token"]}'"
> )